controllers/service.go

```diff
@@ -59,10 +73,33 @@ func servicePorts(es *egressv1.ExternalService) (ports []corev1.ServicePort) {
 	return
 }
 
-func service(es *egressv1.ExternalService) *corev1.Service {
+func service(es *egressv1.ExternalService, ready bool, current *corev1.Service) *corev1.Service {
 	l := labels(es)
-	if es.Spec.HijackDns {
+	switch {
+	// Easy case; if hijacking is disabled, don't hijack
+	case !es.Spec.HijackDns:
+		l["egress.monzo.com/hijack-dns"] = "false"
+
+	// Easy case: pods are ready
+	case ready:
 		l["egress.monzo.com/hijack-dns"] = "true"
+
+	// Creation, not ready - go to waiting state. We'll get another reconcile event when
+	// the ready count changes
+	case current == nil:
+		l["egress.monzo.com/hijack-dns"] = "waiting-for-pods"
+
+	// Enablement, not ready - go to waiting state
+	case current.Labels["egress.monzo.com/hijack-dns"] == "false":
```

Comment:
Can this be `false` while `es.Spec.HijackDns` is `true`?

Reply:
yes if we just turned it on!